src/services/provisioningService.ts

```diff
@@ -0,0 +1,145 @@
+                        { name: "Authorization", value: accessToken }
+                    ],
+                    body: JSON.stringify(contentItem)
+                };
+
+                const response = await this.httpClient.send(request);
+
+                if (response.statusCode >= 400) {
+                    throw new Error("Unable to setup website.");
+                }
+            }
+            this.router.navigateTo("/");
+            this.viewManager.setHost({ name: "content-host" });
+            this.viewManager.showToolboxes();
+        }
+        catch (error) {
+            throw error;
+        }
+    }
+
+    private async cleanupContent(): Promise<void> {
+        const settings = await this.settingsProvider.getSettings();
+        let managementApiUrl = settings["managementApiUrl"];
+        if (!managementApiUrl) {
+            throw new Error(`Management API URL ("managementApiUrl") setting is missing in configuration file.`);
+        }
+        managementApiUrl = Utils.ensureUrlArmified(managementApiUrl);
+        const url = `${managementApiUrl}/contentTypes?api-version=2018-06-01-preview`;
+        const accessToken = this.authenticator.getAccessToken();
+        const request: HttpRequest = {
+            url: url,
+            method: "GET",
+            headers: [
+                { name: "If-Match", value: "*" },
+                { name: "Content-Type", value: "application/json" },
+                { name: "Authorization", value: accessToken }
+            ],
+        };
+        const response = await this.sendRequest(request);
+        const contentTypes = Object.values(response["value"]);
+        for (const contentType of contentTypes) {
+            const contentTypeName = contentType["name"];
+            const curReq: HttpRequest = {
+                url: `${managementApiUrl}/contentTypes/${contentTypeName}/contentItems?api-version=2018-06-01-preview`,
+                method: "GET",
+                headers: [
+                    { name: "If-Match", value: "*" },
+                    { name: "Content-Type", value: "application/json" },
+                    { name: "Authorization", value: accessToken }
+                ],
+            };
+            const itemsResponse = await this.sendRequest(curReq);
+            const items = Object.values(itemsResponse["value"]);
+            for (const item of items) {
+                const itemReq: HttpRequest = {
+                    url: `${managementApiUrl}${item["id"]}?api-version=2018-06-01-preview`,
+                    method: "DELETE", 
+                    headers: [
+                        { name: "If-Match", value: "*" },
+                        { name: "Content-Type", value: "application/json" },
+                        { name: "Authorization", value: accessToken }
+                    ],
+                };
+                const response = await this.httpClient.send(itemReq);
```

Comment:
wrap http requests with try and catch
and use existing function **sendRequest**